models/issue_milestone.go

```diff
@@ -181,7 +181,8 @@ func (milestones MilestoneList) getMilestoneIDs() []int64 {
 // GetMilestonesByRepoID returns all milestones of a repository.
 func GetMilestonesByRepoID(repoID int64) (MilestoneList, error) {
 	miles := make([]*Milestone, 0, 10)
-	return miles, x.Where("repo_id = ?", repoID).Asc("deadline_unix").Find(&miles)
+	return miles, x.Where("repo_id = ? AND is_closed = ?", repoID, 0).
```

Comment:
There is only one place to reference this function?

Reply:
Another reference is https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/milestone.go#L38. The change will affect that results.

Reply:
@lunny I think It depends on expected behaviour... Should the api return both closed and opened milestones or just opened ones..

But whatever your choice, would be interested in fixing them

Reply:
Or should I create ` GetOpenedMilestonesByRepoID(repoID int64)` ( names suggestion ???) so we can leave this one as it was ( keeping api backwards compatibility ).. While moving the `is_closed = 0` to the new function and use it in the issues page

Reply:
https://developer.github.com/v3/issues/milestones/#list-milestones-for-a-repository. It seems github's default milestones' list is opened. :( I think we should make a break change and follow github's API, both change the API docs.

Reply:
Awesome then.... a break change means `kind/breaking` label ?`...  

API docs ? You mean swagger . It doesn't currently show a closed milestone in the response so we should not be needing a change ?

Reply:
Yes, I will add a `kind/breaking` label on this PR and I mean https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/milestone.go#L17 and https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/milestone.go#L21 's comments should be changed by this PR from `all of a repository's milestones` to `opened milestones of a repository`.

Reply:
Done @lunny

Reply:
@adelowo maybe you missed comments at https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/milestone.go#L21 and https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/milestone.go#L17

Reply:
Done @lunny ... Thanks